### PR TITLE
Fix tasty reflect singletons

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

File: library/src/scala/tasty/util/ShowSourceCode.scala [hunk 1]
```diff
@@ -621,6 +622,7 @@ class ShowSourceCode[T <: Tasty with Singleton](tasty0: T) extends Show[T](tasty
 
       case TypeTree.Singleton(ref) =>
         printTree(ref)
+        this += ".type"
```
Comment (liufengyun) [inline library/src/scala/tasty/util/ShowSourceCode.scala]: How Constant types are handled?

Comment (nicolasstucki) [reply library/src/scala/tasty/util/ShowSourceCode.scala]: They are handled by `printConstant`.  I will check that they are printed correctly.

Comment (nicolasstucki) [reply library/src/scala/tasty/util/ShowSourceCode.scala]: @liufengyun fixed the literal constant singleton types  to not print `".type"`

**Comment by liufengyun** [comment]
LGTM

**Pull request merged**